### PR TITLE
add PingFederate connector documentation

### PR DESCRIPTION
**Pull request by tharindu1st**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/administer/key-managers/configure-pingfederate-connector.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
+# Configure PingFederate as a Key Manager
+
+WSO2 API Manager supports multiple Key Managers. As a result, WSO2 API Manager is prepacked with an inbuilt resident Key Manager and with the use of connectors it is capable of supporting any authorization server as a Key Manager.
```
Comment (bhathiya) [inline en/docs/administer/key-managers/configure-pingfederate-connector.md:3]:
```suggestion
WSO2 API Manager supports multiple Key Managers at the same time. As a result, WSO2 API Manager is prepacked with an inbuilt resident Key Manager, and with the use of connectors, it is capable of supporting any authorization server as a Key Manager.
```

File: en/docs/administer/key-managers/configure-pingfederate-connector.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+Therefore, WSO2 API Manager can connect PingFederate out-of-the-box using the [WSO2 API-M PingFederate Connector](https://github.com/wso2-extensions/apim-km-pingfederate).
```
Comment (bhathiya) [inline en/docs/administer/key-managers/configure-pingfederate-connector.md:5]:
```suggestion
As a part of that, WSO2 API Manager can connect PingFederate out-of-the-box using the [WSO2 API-M PingFederate Connector](https://github.com/wso2-extensions/apim-km-pingfederate).
```

File: en/docs/administer/key-managers/configure-pingfederate-connector.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+## Step 2 - Configure Keycloak
+
+1. Import PingFederate certificate into the WSO2 API Manager truststore.
+
+    - The default PingFederate keystore can found from `SSL Server Certificates` under `Certificate & Key Management`.
+
+     [![SSL Server Certificate]({{base_path}}/assets/img/administer/ping-federate-ssl-certificate.png)]({{base_path}}/assets/img/administer/ping-federate-ssl-certificate.png)
+
+2. Export the certificate from SSL server certificates.
```
Comment (bhathiya) [inline en/docs/administer/key-managers/configure-pingfederate-connector.md:38]: 2 and 3 should be sub-points if 1 right?

File: en/docs/administer/key-managers/configure-pingfederate-connector.md (hunk 1 above)
Comment (isharac) [inline en/docs/administer/key-managers/configure-pingfederate-connector.md:3]: We have to mention that this OOTB support available only through wum (timestamp)

File: en/mkdocs.yml [hunk 4]
```diff
@@ -721,4 +722,4 @@ extra:
         - type: linkedin
           link: https://www.linkedin.com/company/wso2
     site_version: 3.2.0
-    base_path: https://apim.docs.wso2.com/en/3.2.0
+    base_path: http://localhost:8000
```
Comment (bhathiya) [inline en/mkdocs.yml:725]: revert this.

**Review by bhathiya — CHANGES_REQUESTED** [review]
added suggestions

File: en/docs/administer/key-managers/configure-pingfederate-connector.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+    [![PingFederate add client]({{base_path}}/assets/img/administer/pingfederate-client-create-1.png)]({{base_path}}/assets/img/administer/pingfederate-client-create-1.png)
+    [![PingFederate add client]({{base_path}}/assets/img/administer/pingfederate-client-create-2.png)]({{base_path}}/assets/img/administer/pingfederate-client-create-2.png)
+
+3. Generate and Store the CLIENT SECRET.
+
+     [![Keycloak client secret]({{base_path}}/assets/img/administer/pingfederate-client-secret.png)]({{base_path}}/assets/img/administer/pingfederate-client-secret.png)
+
+
+## Step 2 - Configure Keycloak
```
Comment (isharac) [inline en/docs/administer/key-managers/configure-pingfederate-connector.md:30]: Keycloak?

File: en/mkdocs.yml (hunk 4 above)
Comment (isharac) [inline en/mkdocs.yml:725]: need to revert this change

**Review by isharac — CHANGES_REQUESTED** [review]

File: en/docs/administer/key-managers/configure-pingfederate-connector.md (hunk 1 above)
Comment (tharindu1st) [reply en/docs/administer/key-managers/configure-pingfederate-connector.md:3]: this need to mention the wum timestamp

**Review by isharac — APPROVED** [review]

**Review by bhathiya — APPROVED** [review]

File: en/docs/administer/key-managers/configure-pingfederate-connector.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+          <tr class="even">
+            <td>Self Validate JWT</td>
+            <td>The kid value is used to validate the JWT token signature. If the kid value is not present, `gateway_certificate_alias` is used.</td>
+            <td>Optional</td>
+          </tr>
+          <tr class="odd">
+            <td>Use introspect</td>
+            <td>The JWKS endpoint is used to validate the JWT token signature.
+            If this option is used to validate the tokens it is mandatory to add a Token Handling Option.
+                For the Okta it should be <b>JWT</b> and it is required to specify a claim mapping as a unique identifier.
```
Comment (Mariangela) [inline en/docs/administer/key-managers/configure-pingfederate-connector.md:252]: Is it correct to mention Okta here when we are speaking about Ping Federate?

File: en/docs/administer/key-managers/configure-pingfederate-connector.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+            <td>CUSTOM</td>
+            <td>The token matching a custom pattern is validated.</td>
+            <td>Optional</td>
+          </tr>
+          <tr class="even">
+            <td><b>Claim Mappings</b></td>
+            <td>Local and remote claim mapping.</td>
+            <td>Optional</td>
+          </tr>
+          </table>
```
Comment (Mariangela) [inline en/docs/administer/key-managers/configure-pingfederate-connector.md:283]: How come we do not have a section on "generate keys using the PingFederate key manager" similar to what we have for okta [1], Auth0, keycloak etc.?

[1] https://apim.docs.wso2.com/en/latest/administer/key-managers/configure-okta-connector/#step-3-generate-keys-using-the-okta-key-manager

File: en/mkdocs.yml [hunk 8]
```diff
@@ -719,4 +720,4 @@ extra:
...
```
Comment (Mariangela) [inline en/mkdocs.yml]: This change is incorrect. The base_path should be "base_path: https://apim.docs.wso2.com/en/3.2.0"

**Review by Mariangela — CHANGES_REQUESTED** [review]
@tharindu1st , please note the comments that I have added and also please make sure to pull my changes to your local repo when addressing the review comments so that my changes do not get deleted.

File: en/mkdocs.yml (hunk 8 above)
Comment (tharindu1st) [reply en/mkdocs.yml]: fixed

File: en/docs/administer/key-managers/configure-pingfederate-connector.md (hunk 7 above)
Comment (tharindu1st) [reply en/docs/administer/key-managers/configure-pingfederate-connector.md:283]: fixed

**Review by Mariangela — APPROVED** [review]
